dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/CheckpointImpl.java

```diff
@@ -153,6 +160,7 @@ void checkpoint(Phase phase) throws CheckpointFailed {
             throw new CheckpointFailed(Type.SNAPSHOT_FAILED, "Failed to do checkpoint.", e, 0);
         }
         restore(phase, checkpointHooks);
+        checkpointCalled.set(true);
```

Comment:
No need for this to be here if you use compareAndSet

Reply:
Fixed up and added a test